packages/http/src/mocker/negotiator/InternalHelpers.ts

```diff
@@ -23,18 +26,29 @@ export function hasContents(v: IHttpOperationResponse): v is PickRequired<IHttpO
 }
 
 export function findBestHttpContentByMediaType(
-  response: PickRequired<IHttpOperationResponse, 'contents'>,
-  mediaType: string[]
+  contents: IMediaTypeContent[],
+  mediaTypes: string[]
 ): Option<IMediaTypeContent> {
-  const bestType = accepts({
-    headers: {
-      accept: mediaType.join(','),
-    },
-  }).type(response.contents.map(c => c.mediaType));
+  const bestType: string | false = accepts({ headers: { accept: mediaTypes.join(',') } }).type(
+    contents.map(c => c.mediaType)
+  );
 
   return pipe(
-    response.contents,
-    findFirst(content => content.mediaType === bestType)
+    bestType,
+    O.fromPredicate((bestType): bestType is string => !!bestType),
+    O.chain(bestType => findFirst<IMediaTypeContent>(content => content.mediaType === bestType)(contents)),
+    O.alt(() =>
+      // Since media type parameters are not standardised (apart from the quality value), we're going to try again ignoring them all but q.
+      pipe(
+        mediaTypes
+          .map(mt => contentType.parse(mt))
+          .filter(mt => Object.keys(mt.parameters).filter(k => k !== 'q').length > 0)
```

Comment:
```suggestion
          .filter(({ parameters }) => Object.keys(parameters).find(k => k !== 'q'))
```
or for more clarity:
```suggestion
          .filter(({ parameters }) => !!Object.keys(parameters).find(k => k !== 'q'))
```

Reply:
Yeah good point. @karol-maciaszek how about this: bbb4c73